### PR TITLE
Feat:"Added Auth"

### LINKED ISSUE #9: feat: Signup and Login Page

### PR DESCRIPTION
**Pull request by sadathshariff**
Added Auth Context
Implemented Signup, Login  and Logout

### CODE REVIEW

File: src/components/ProductDetails/ProductDetail.jsx [hunk 1]
```diff
@@ -31,10 +42,15 @@ export const ProductDetail = () => {
               </p>
             </div>
             <div className="product-details-btns">
-              <button className="btn btn-outline-primary">
+              <button
+                className="btn btn-outline-primary"
+                onClick={() => console.log("add to Wishlist")}
```
Comment (nishasen) [inline src/components/ProductDetails/ProductDetail.jsx]: remove the console.logs later

File: src/pages/Auth/Login.jsx [hunk 2]
```diff
@@ -1,3 +1,108 @@
...
+import { Button } from "../../components";
+import axios from "axios";
+import { useAuth } from "../../contexts";
 export const Login = () => {
-  return <div>Login Page</div>;
+  const initialData = {
+    email: "",
+    password: "",
+  };
+  const testCredentials = {
```
Comment (nishasen) [inline src/pages/Auth/Login.jsx]: use this for guest login, it will be helpful

File: src/pages/Auth/auth.css [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+.login-card {
+  margin: 4rem auto;
+}
+.signup,
+.login {
+  max-width: 28rem;
+  margin: auto;
+  padding: 1rem;
+  background-color: var(--off-white);
+  box-shadow: rgba(0, 0, 0, 0.07) 0px 1px 1px, rgba(0, 0, 0, 0.07) 0px 2px 2px,
```
Comment (nishasen) [inline src/pages/Auth/auth.css]: try storing the big values inside a variable, and use it here

**Review by nishasen — APPROVED** [review]
Suggested small changes, your code looks clean and I like how you used the userToken.

File: src/pages/Auth/Login.jsx (hunk 2 above)
Comment (sadathshariff) [reply src/pages/Auth/Login.jsx]: I'm using that for guest Login

File: src/pages/Auth/auth.css (hunk 3 above)
Comment (sadathshariff) [reply src/pages/Auth/auth.css]: Yeah Will do that

**Review by raman1999 — APPROVED** [review]
clean code written and good breakdown of bigger components done,

**Pull request merged**